iree/turbine/kernel/wave/codegen/handlers.py

```diff
@@ -363,23 +371,32 @@ def handle_generic_binary(emitter: WaveEmitter, node: fx.Node):
                 lhs, rhs = node.args
             except ValueError as e:
                 raise ValidationError("Malformed arguments") from e
-            lhs = cast_py_value(emitter, lhs)
-            rhs = cast_py_value(emitter, rhs)
-
-            if lhs.ir_value.type != rhs.ir_value.type:
+            lhs = cast_py_value(emitter, lhs).ir_value
+            rhs = cast_py_value(emitter, rhs).ir_value
+
+            # Handle special scalar/rank-0 cases where lhs/rhs may be
+            # Dtype, vector<Dtype>, or vector<1xDtype>.
+            arg_ranks = [get_rank(arg.type) for arg in (lhs, rhs)]
+            if min(arg_ranks) != max(arg_ranks) and max(arg_ranks) <= 1:
```

Comment:
Since this is a binary op, can you not do
```
(arg_ranks[0] != arg_ranks[1])  and max(arg_ranks) <= 1
```
Do you need to compute the min and max?

Reply:
this makes sense :)

Reply:
done!